### ISSUE TITLE
Major roads should be rendered from world basemap at z>11, but on special background

### ISSUE THREAD

**Issue by polarbearing** [osmandapp/Osmand#2818]
From z12 and higher, world basemap data is not being rendered anymore. From reading #1777 I understand that this is intended. However it leads to a situation that when driving along a calculated route (for which worldmap data are considered) goes through a region that has no detail data loaded, the roads suddenly disappear, and a device fault is assumed. I understand that the basemap data might have a simplified geometry. 

I suggest one or more  of the following improvements:
- render major roads (and other available data) from worldmap in higher zooms, 
- render a special background pattern (warning colour, hatching, or even the words "base map") behind the roads
- render such special background only, as a warning to zoom back
- adjust the automatic zoom level to perform that zooming back automatically while driving.

In the screenshot below it is still obvious that you are running out of data, but in a higher zoom the map gets suddenly blank:

![screenshot_2016-07-13-19-28-26](https://cloud.githubusercontent.com/assets/8266355/16814502/65092316-4936-11e6-95a9-3dfe4659c851.png)
![screenshot_2016-07-13-19-28-15](https://cloud.githubusercontent.com/assets/8266355/16814522/7ddf7ed0-4936-11e6-904e-434ed9b9a8e9.png)

**Comment by polarbearing** [comment]
Yes I just experienced that on another trip through basemap land. What happens is that a route is calculated _around_ the basemap areas. Once you are in this area, the route is lost and fails to recalculate since the current location is far from reachable roads. Since autozoom tries to go into detail level, nothing is rendered anymore and the app appears to be broken. 

Thus suggestions 2-4 above remain valid. 

How much would the basemap be increased if it contained routing data for secondary roads and above?

**Comment by vshcherb** [comment]
It is a known issue, the suggestions are valid in case we manage to indicate that area is a base map area as a warning, so people will pay attention to download it.

**Comment by polarbearing** [comment]
How difficult would it be to render the blank area with no detail map in "fixme"-pink, maybe with a label "No Details"? That would also make routing issues more obvious (easier to analyse) where the motorway goes just a few miles through unloaded territory. 

Also, since the zoom level ist not displayed constantly, using the base map should be better recognisable. Maybe showing a world icon.

Example below, bypassing the A9 via A14 since Saxonia details are unloaded.

![screenshot_20170528-bypassing_saxonia](https://cloud.githubusercontent.com/assets/8266355/26578955/28209c6a-4532-11e7-9782-fe2ae70086f8.png)

**Comment by sonora** [comment]
As a side note: The original issue that nothing is rendered from the world base map at zooms closer than 11 only exists in areas where a portion of the map screen is covered by a standard (regional) map while another portion remains blank because of a missing standard map there (even though the world basemap is present and could cover this missing region). If, on the other hand, you zoom into an area where the **entire screen** is not covered by any regional map you have on your device, the data from the world basemap is always displayed ok even at zooms closer than 11.
